### PR TITLE
Use context pattern

### PR DESCRIPTION
**Pull request by vladbarosan**

### CODE REVIEW

File: src/Templates/MethodTemplate.cshtml [hunk 1]
```diff
@@ -162,9 +161,7 @@ func (client @(Model.Owner)) @(Model.PreparerMethodName)(@(Model.MethodParameter
 </text>
 }
 
-    @(Model.IsLongRunningOperation()
-        ? "return preparer.Prepare(&http.Request{Cancel: cancel})"
-        : "return preparer.Prepare(&http.Request{})")
+    return preparer.Prepare((&http.Request{}).WithContext(ctx))
 }
```
Comment (mcardosos) [inline src/Templates/MethodTemplate.cshtml]: I would add a check for nil contexts

Comment (vladbarosan) [reply src/Templates/MethodTemplate.cshtml]: withContext will Panic if the context is `nil`. Is there any point in having an extra check and presumably to return an error ?

Comment (mcardosos) [reply src/Templates/MethodTemplate.cshtml]: I just wonder if it is a good idea to set the context to `context.Background()` if the user is passing a nil context, just for it not to panic.

**Comment by vladbarosan** [comment]
Not sure why the CI is failing today. seems to be for some breaking changes ? those where yesterday also but I don't remember failing. Only thing I did today was the resolve conflicts after @jhendrixMSFT 's changes

**Comment by vladbarosan** [comment]
Ok it seems that this PR #39 made the 2 params in the generated test code to be on the same line. Not sure why is that since previously it was on separate lines and for me also it puts it on the same line. @jhendrixMSFT any ideas why that change ? also do we want to revert or stay the same ?

**Comment by olydis** [comment]
> win32-x64 restart

**Comment by vladbarosan** [comment]
> win32-x64 restart

**Pull request merged**